lib/ansible/modules/windows/win_uri.ps1

```diff
@@ -19,43 +19,44 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
-$params = Parse-Args $args;
-
-$result = New-Object psobject @{
-    win_uri = New-Object psobject
-}
+$ErrorActionPreference = "Stop"
 
 # Functions ###############################################
 
 Function ConvertTo-SnakeCase($input_string) {
     $snake_case = $input_string -csplit "(?<!^)(?=[A-Z])" -join "_"
-    $snake_case = $snake_case.ToLower()
-
-    return $snake_case
+    return $snake_case.ToLower()
 }
 
 # Build Arguments
-$webrequest_opts = @{}
-
-$url = Get-AnsibleParam -obj $params -name "url" -failifempty $true
-$method = Get-AnsibleParam -obj $params "method" -default "GET"
-$content_type = Get-AnsibleParam -obj $params -name "content_type"
-$headers = Get-AnsibleParam -obj $params -name "headers"
-$body = Get-AnsibleParam -obj $params -name "body"
-$dest = Get-AnsibleParam -obj $params -name "dest" -type "path" -default $null
-$use_basic_parsing = ConvertTo-Bool (Get-AnsibleParam -obj $params -name "use_basic_parsing" -default $true)
-
-$webrequest_opts.Uri = $url
-Set-Attr $result.win_uri "url" $url
-
-$webrequest_opts.Method = $method
-Set-Attr $result.win_uri "method" $method
-
-$webrequest_opts.ContentType = $content_type
-Set-Attr $result.win_uri "content_type" $content_type
+$params = Parse-Args $args -supports_check_mode $true
+
+$url = Get-AnsibleParam -obj $params -name "url" -type "string" -failifempty $true
+$method = Get-AnsibleParam -obj $params "method" -type "string" -default "GET" -validateset "GET","POST","PUT","HEAD","DELETE","OPTIONS","PATCH","TRACE","CONNECT","REFRESH"
+$content_type = Get-AnsibleParam -obj $params -name "content_type" -type "string"
+# TODO: Why is this not a normal dictionary ?
```

Comment:
Because I'm generally a Linux admin. This and the win_robocopy module are pretty much the only powershell I've written. If there's a better way to do anything in this module please go for it. My only reasoning for not doing it that way to begin with was ignorance.